src/components/Menu/Menu.tsx

```diff
@@ -133,7 +141,7 @@ const Menu: FC<MenuProps> = ({ children, menuButtonRef }) => {
                   aria-expanded={open}
                 />
               </Box>
-              <Flex
+              <NavMenuList
```

Comment:
Did you try  role="navigation" - If it works could save on creating the extra component ?
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/navigation_role

Reply:
I did consider that but I think it's better / more a11y friendly to use semantic HTML where possible, and it's easy enough to do here!